### PR TITLE
[ntuple] Add `Internal::MergeModels`

### PR DESCRIPTION
**Pull request by enirolf**
The addition of this function is in anticipation for horizontal joining of RNTuples, which will be handled in the context of the `RNTupleProcessor` that is currently under development (see also #15084).

## Checklist:

- [x] tested changes locally
- [x] updated the docs (if necessary)

### CODE REVIEW

**Review by vepadulano — APPROVED** [review]
LGTM! Failures are unrelated (TMVA)

File: tree/ntuple/v7/src/RNTupleModel.cxx [hunk 1]
```diff
@@ -33,6 +33,33 @@ std::uint64_t GetNewModelId()
...
+
+   auto newModel = left.Clone();
+   newModel->Unfreeze();
+
+   for (const auto &f : right.GetFieldZero().GetSubFields()) {
+      std::string fieldName;
+      if (rightFieldPrefix.empty()) {
+         fieldName = f->GetFieldName();
+      } else {
+         fieldName = std::string(rightFieldPrefix) + ":" + f->GetFieldName();
```
Comment (jblomer) [inline tree/ntuple/v7/src/RNTupleModel.cxx:51]: Shouldn't this be a period (`.`) as a separator?

**Review by jblomer — APPROVED** [review]
LGTM, just one question on the prefix separator.

File: tree/ntuple/v7/src/RNTupleModel.cxx (hunk 1 above)
Comment (enirolf) [reply tree/ntuple/v7/src/RNTupleModel.cxx:51]: Ideally yes, but this clashes with `RFieldBase::EnsureValidFieldName`. Let me see if I can come up with a workaround that's not too hacky.

**Review by jblomer — APPROVED** [review]

**Pull request merged**